q2_feature_classifier/classifier.py

```diff
@@ -104,20 +106,41 @@ def fit_classifier(reference_reads: DNAIterator, reference_taxonomy: pd.Series,
 
 
 plugin.methods.register_function(
-    function=fit_classifier,
+    function=fit_classifier_sklearn,
     inputs={'reference_reads': FeatureData[Sequence],
             'reference_taxonomy': FeatureData[Taxonomy]},
     parameters={'classifier_specification': Str},
     outputs=[('classifier', TaxonomicClassifier)],
-    name='Train a scikit-learn classifier.',
+    name='Train an almost arbitrary scikit-learn classifier',
     description='Train a scikit-learn classifier to classify reads.'
 )
 
 
-def classify(reads: DNAIterator, classifier: Pipeline,
-             chunk_size: int=262144, n_jobs: int=1,
-             pre_dispatch: str='2*n_jobs', confidence: float=-1.
-             ) -> pd.DataFrame:
+def _autodetect_read_direction(reads, classifier, n=100,
+                               force_read_direction=None):
+    if force_read_direction == 'forward':
+        return reads
+    if force_read_direction == 'reverse':
+        return (r.reverse_complement() for r in reads)
+    reads = iter(reads)
+    first_n_reads = list(islice(reads, n))
+    result = list(zip(*predict(first_n_reads, classifier, confidence=0.)))
+    _, _, forward_confidence = result
+    reversed_n_reads = [r.reverse_complement() for r in first_n_reads]
+    result = list(zip(*predict(reversed_n_reads, classifier, confidence=0.)))
+    _, _, reverse_confidence = result
+    if sum(forward_confidence) > sum(reverse_confidence):
```

Comment:
I don't have a good feeling for which will do better, but since confidence is bound by 1.0 (I wasn't thinking about that), maybe sum does make more sense.